### PR TITLE
Support Dockerfile HEALTHCHECK of agent

### PR DESCRIPTION
**Pull request by cyastella**
### Summary

Address #800. Previously, ECS does not have a Healthcheck directive for itself. This PR add the healthcheck in the dockerfile to enable detect the health status of ECS agent.
### Implementation details

Enable ECS agent healthcheck flag 
Add the healthcheck command in the dockerfile. 
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by samuelkarp** [comment]
Instead of mounting in another binary, can you extend the existing `agent` binary inside the container to be able to report health of the container?

**Comment by petderek** [comment]
`/agent --healthcheck` would be appropriate, and makes it a bit more extensible.

File: agent/app/agent.go [hunk 1]
```diff
@@ -196,6 +199,15 @@ func (agent *ecsAgent) printECSAttributes() int {
 	return exitcodes.ExitSuccess
 }
 
+func (agent *ecsAgent) runHealthcheck() int {
+	_, err := http.Get("http://localhost:51678/v1/metadata")
+	if err != nil {
+		seelog.Warnf("Unable to obtain healthcheck: %v", err)
```
Comment (fenxiong) [inline agent/app/agent.go]: How about something like "Health check failed with error: ..."

File: agent/app/args/flag.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ const (
 	licenseUsage             = "Print the LICENSE and NOTICE files and exit"
 	blacholeEC2MetadataUsage = "Blackhole the EC2 Metadata requests. Setting this option can cause the ECS Agent to fail to work properly.  We do not recommend setting this option"
 	windowsServiceUsage      = "Run the ECS agent as a Windows Service"
+	healthcheckServiceUsage  = "Run the agent healthcheck info"
```
Comment (fenxiong) [inline agent/app/args/flag.go]: maybe just "Run the agent healthcheck"..

File: agent/app/agent.go (hunk 1 above)
Comment (cyastella) [reply agent/app/agent.go]: yeah, will change it

File: agent/app/healthcheck.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (fenxiong) [inline agent/app/healthcheck.go]: nit: -2019

Comment (suneyz) [reply agent/app/healthcheck.go]: `Copyright 2019 Amazon.com` since it's a new file?

File: agent/app/healthcheck.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2014-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (yumex93) [inline agent/app/healthcheck.go]: nit: the year can be just 2019

**Pull request merged**